lib/telegraf/rack.rb

```diff
@@ -100,7 +101,8 @@ def extract_request_start(env)
       return unless env.key?('HTTP_X_REQUEST_START')
 
       if (m = HEADER_REGEX.match(env['HTTP_X_REQUEST_START']))
-        ::Time.at(m[1].to_f).utc
+        value = m[2].nil? ? "#{m[1][0, 10]}.#{m[1][10, 13]}" : m[1]
+        ::Time.at(value.to_f).utc
```

Comment:
This assumes that the seconds in a UNIX timestamp will be exactly 10 digits long, does it? Everything else is taken as a sub-second precision?

Nice solution, but that will fail on Saturday, November 20, 2286 5:46:40 PM when the UNIX timestamp grows by one digit :joy:. Probably not an issue yet.

Reply:
Yeah, I'm sure we'll be good for the next 250 years 😉